app/parse_class.py

```diff
@@ -0,0 +1,108 @@
+import csv
+from dataclasses import dataclass
+from typing import List
+from urllib.parse import urljoin
+
+import requests
+from bs4 import BeautifulSoup
+from selenium import webdriver
+from selenium.webdriver import ActionChains
+from selenium.webdriver.common.by import By
+
+
+@dataclass
+class Product:
+    title: str
+    description: str
+    price: float
+    rating: int
+    num_of_reviews: int
+
+
+class ParseClass:
+
+    BASE_URL = "https://webscraper.io/"
+    HOME_URL = urljoin(BASE_URL, "test-sites/e-commerce/more/")
+
+    def __init__(self, gadget_type: str = None) -> None:
+        self.gadget_type = gadget_type
+        self.file_name = "home.csv" if not gadget_type else None
```

Comment:
it's better to use vice-verse condition to increase readability
None if gadget_type else home.csv